### ISSUE TITLE
Adjust default display order for browsing and for displaying filtered results

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/pressbooks-book-directory-fe#403]
As a visitor to the Directory, I'd like to see recently updated books, so that the browsing experience continually presents me with new and interesting books. When I filter my results or perform a search, however, I'd like to see the results sorted by relevancy.

We should do the following: 
1. change the default 'browsing' display order for visitors to the directory to show them results using the last updated replica index
1. whenever a user performs a search and/or applies a filter, automatically change the default display to show them the relevance sort order. Once they've performed a query or applied a filter, they should continue to be permitted to change the display order to either of the other sort order options: 'last updated' or 'title A-Z'. Once they've changed this manually, it should be 'sticky' until the query is reset or all filters are cleared, at which point we should revert back to 'browsing' display order.

**Comment by richard015ar** [comment]
I've been working around 6 hours in this task and I tried 2 different solutions.
Most interesting one so far, is to listening changes in the query (URL GET parameters) in main.js. If there is some query present, this means some filter or search was performed and we need to change the index to "relevance". Otherwise if there is not query present in the URL, then we must use the "last updated" index.
I doing it in the routing change listener:
```js
router.beforeEach((to, from, next) => {
  if (to.query[store.state.SClient.searchParameters.aliases.sortedBy] && store.state.SClient.resetMainIndex) {
    // Sort By feature using URL
    const indexesOrderedByMap = store.state.SClient.availableIndexes.reduce((index, item) => {
      return {
        ...index,
        [item.orderedBy]: item.value
      };
    }, {});
    store.commit('setSortedBy', to.query[store.state.SClient.searchParameters.aliases.sortedBy]);
    store.commit('setMainIndex', indexesOrderedByMap[to.query[store.state.SClient.searchParameters.aliases.sortedBy]]);
  }
  if ( Object.keys(to.query).length === 0) {
    store.state.SClient.searchParameters.sortedBy = 'updated';
    store.state.SClient.availableIndexes[0].default = false;
    store.state.SClient.availableIndexes[1].default = true;
    store.state.SClient.indexName = import.meta.env.VITE_ALGOLIA_INDEX_LAST_UPDATED_REPLICA;
  } else {
    store.state.SClient.searchParameters.sortedBy = 'relevance';
    store.state.SClient.availableIndexes[1].default = false;
    store.state.SClient.availableIndexes[0].default = true;
    store.state.SClient.indexName = import.meta.env.VITE_ALGOLIA_INDEX;
  }
...
```
It works as expected, the problem is t[he main instant search component](https://www.algolia.com/doc/api-reference/widgets/instantsearch/vue/) is not updated properly, even if we change the `indexName` parameter. Looks like we need to refresh or update the component in some way.
I was investigating this, and I also did the same research for the SortBy select feature, which does not work in another way than using [the sort by component](https://www.algolia.com/doc/api-reference/widgets/sort-by/vue/), which trigger the action only when a new item is selected inside the component by using the `refine` method.

In other words, I didn't find a way to **refresh** or **update** the main algolia component and read the new index defined in `indexName` parameter.
Other parameters are being reload properly (like the `facetFilters` one for example). But looks like change index is something cached or something that is not easy to reload.
If we reload the page, then the correct index will be loaded as expected. Which means, the main algolia component is loaded properly at the beginning.

I will move this task to **In progress** sprint board column since we are going to work in the Export routine hackathon.
